code/game/objects/items/weapons/implants/exile2.dm

```diff
@@ -1,39 +1,125 @@
-/**
-* Exile Implants 2.0
-*
-* Since away missions are fucked, here's an alternative implementation.
-*
-* Instead of confining someone within an away mission, this locks someone to the asteroid.
-*/
+
+//There now it's fun
+
 /obj/item/weapon/implanter/exile
 	name = "implanter-exile"
 
 /obj/item/weapon/implanter/exile/New()
-	src.imp = new /obj/item/weapon/implant/exile( src )
+	src.imp = new /obj/item/weapon/implant/exile(src)
 	..()
 	update()
 	return
 
-
 /obj/item/weapon/implant/exile
 	name = "exile"
-	desc = "Prevents you from returning from the asteroid"
+	desc = "Prevents returning to where you were implanted"
+	var/illegalZ = null
+	var/mob/living/theExile = null
```

Comment:
implants already have a var to track the mob they're implanted into, called `imp_in`, which is a terrible name but it's probably better than using a special snowflake var for the same purpose.

Reply:
I can't change that, the implant requires typecasting to work. Won't even register as implanted in the person if I use imp_in. Even the previous version of the implant had a typecast variable to replace imp_in.